### PR TITLE
adding option to build the new g2c functionality

### LINKED ISSUE #446: add option to build new g2c API functions
Add build option BUILD_G2C which will cause the new g2c_ API to be built. This option will be off by default until the 2.0.0 release.

### PR DESCRIPTION
**Pull request by edwardhartnett**
Fixes #446 

This adds an option BUILD_G2C, which is OFF by default.

Turning it on causes all the new g2c functionality to be included in the build. The BUILD_G2C option will be ON by default starting with the 2.0.0 release.

### CODE REVIEW

File: .github/workflows/Spack.yml [hunk 1]
```diff
@@ -60,8 +61,8 @@ jobs:
         spack load g2c
         if [[ "${{ matrix.variants }}" =~ "libs=shared" ]]; then suffix="so" ; else suffix="a"; fi
         ls $G2C_LIB | grep -cE '/libg2c\.'$suffix'$'
-        if [[ "${{ matrix.variants }}" =~ "+png"[+~]?$ && "${{ matrix.variants }}" =~ "libs=shared"[+~]?$ ]]; then ldd $G2C_LIB | grep -c libpng; fi
-        if [[ "${{ matrix.variants }}" =~ "+utils"[+~]?$ ]]; then
+        if [[ "${{ matrix.variants }}" == *"+png"* && "${{ matrix.variants }}" == "libs=shared"* ]]; then ldd $G2C_LIB | grep -c libpng; fi
+        if [[ "${{ matrix.variants }}" == *"+utils"* ]]; then
```
Comment (edwardhartnett) [inline .github/workflows/Spack.yml:65]: @AlexanderRichert-NOAA this was not working and I was confused as heck until I read this: https://stackoverflow.com/questions/229551/how-to-check-if-a-string-contains-a-substring-in-bash

Comment (AlexanderRichert-NOAA) [reply .github/workflows/Spack.yml:65]: Okay good to know. In the future I'll use that as the model for the other libraries.

**Comment by edwardhartnett** [comment]
@AlexanderRichert-NOAA or @Hang-Lei-NOAA please review and I will do the 1.8.0 release of g2c.

**Review by AlexanderRichert-NOAA — APPROVED** [review]
Looks good

**Review by Hang-Lei-NOAA — APPROVED** [review]

**Pull request merged**